File: IrrgartenP2/IrrgartenJava/src/irrgartenjava/TestP1.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgartenjava;

/**
 *
 * @author moham
 */
// TestP1.java
// TestP1.java
public class TestP1 {
    public static void main(String[] args) {
        // Prueba de Weapon
        Weapon weapon = new Weapon(2.0f, 5);
        System.out.println("Weapon initial: " + weapon);
        System.out.println("Weapon Attack: " + weapon.attack());
        System.out.println("Weapon after attack: " + weapon);
        for (int i = 0; i < 6; i++) {
            System.out.println("Weapon Attack Attempt " + (i + 1) + ": " + weapon.attack());
        }
        System.out.println("Weapon final: " + weapon);
        System.out.println("Weapon Discard: " + weapon.discard());

        // Prueba de Shield
        Shield shield = new Shield(3.0f, 4);
        System.out.println("\nShield initial: " + shield);
        System.out.println("Shield Protect: " + shield.protect());
        System.out.println("Shield after protect: " + shield);
        for (int i = 0; i < 5; i++) {
            System.out.println("Shield Protect Attempt " + (i + 1) + ": " + shield.protect());
        }
        System.out.println("Shield final: " + shield);
        System.out.println("Shield Discard: " + shield.discard());

        // Prueba de Dice
        System.out.println("\nDice Tests:");
        for (int i = 0; i < 100; i++) {
            System.out.println("Resurrect Player Attempt " + (i + 1) + ": " + Dice.resurrectPlayer());
        }

        System.out.println("Random Position: " + Dice.randomPos(10));
        System.out.println("Who Starts: " + Dice.whoStarts(4));
        System.out.println("Random Intelligence: " + Dice.randomIntelligence());
        System.out.println("Random Strength: " + Dice.randomStrength());
        System.out.println("Weapons Reward: " + Dice.weaponsReward());
        System.out.println("Shields Reward: " + Dice.shieldsReward());
        System.out.println("Health Reward: " + Dice.healthReward());
        System.out.println("Weapon Power: " + Dice.weaponPower());
        System.out.println("Shield Power: " + Dice.shieldPower());
        System.out.println("Uses Left: " + Dice.usesLeft());
        System.out.println("Intensity: " + Dice.intensity(5.0f));
        System.out.println("Discard Element (uses = 3): " + Dice.discardElement(3));
        System.out.println("Discard Element (uses = 0): " + Dice.discardElement(0));
        System.out.println("Discard Element (uses = 5): " + Dice.discardElement(5));

        // Prueba de GameState
        GameState gameState = new GameState("Labyrinth", "Player1", "Monster1", 0, false, "Game started");
        System.out.println("\nGameState:");
        System.out.println("Labyrinth: " + gameState.getLabyrinth());
        System.out.println("Players: " + gameState.getPlayers());
        System.out.println("Monsters: " + gameState.getMonsters());
        System.out.println("Current Player: " + gameState.getCurrentPlayer());
        System.out.println("Is Winner: " + gameState.isWinner());
        System.out.println("Log: " + gameState.getLog());
    }
}

File: IrrgartenP2/irrgartenRuby/test_p1.rb
require_relative 'weapon'
require_relative 'shield'
require_relative 'dice'
require_relative 'game_state'

class TestP1
  def self.main
    # Prueba de Weapon
    weapon = Weapon.new(2.0, 5)
    puts "Weapon initial: #{weapon}"
    puts "Weapon Attack: #{weapon.attack}"
    puts "Weapon after attack: #{weapon}"
    6.times do |i|
      puts "Weapon Attack Attempt #{i + 1}: #{weapon.attack}"
    end
    puts "Weapon final: #{weapon}"
    puts "Weapon Discard: #{weapon.discard}"

    # Prueba de Shield
    shield = Shield.new(3.0, 4)
    puts "\nShield initial: #{shield}"
    puts "Shield Protect: #{shield.protect}"
    puts "Shield after protect: #{shield}"
    5.times do |i|
      puts "Shield Protect Attempt #{i + 1}: #{shield.protect}"
    end
    puts "Shield final: #{shield}"
    puts "Shield Discard: #{shield.discard}"

    # Prueba de Dice
    puts "\nDice Tests:"
    100.times do |i|
      puts "Resurrect Player Attempt #{i + 1}: #{Dice.resurrect_player}"
    end

    puts "Random Position: #{Dice.random_pos(10)}"
    puts "Who Starts: #{Dice.who_starts(4)}"
    puts "Random Intelligence: #{Dice.random_intelligence}"
    puts "Random Strength: #{Dice.random_strength}"
    puts "Weapons Reward: #{Dice.weapons_reward}"
    puts "Shields Reward: #{Dice.shields_reward}"
    puts "Health Reward: #{Dice.health_reward}"
    puts "Weapon Power: #{Dice.weapon_power}"
    puts "Shield Power: #{Dice.shield_power}"
    puts "Uses Left: #{Dice.uses_left}"
    puts "Intensity: #{Dice.intensity(5.0)}"
    puts "Discard Element (uses = 3): #{Dice.discard_element(3)}"
    puts "Discard Element (uses = 0): #{Dice.discard_element(0)}"
    puts "Discard Element (uses = 5): #{Dice.discard_element(5)}"

    # Prueba de GameState
    game_state = GameState.new("Labyrinth", "Player1", "Monster1", 0, false, "Game started")
    puts "\nGameState:"
    puts "Labyrinth: #{game_state.labyrinth}"
    puts "Players: #{game_state.players}"
    puts "Monsters: #{game_state.monsters}"
    puts "Current Player: #{game_state.current_player}"
    puts "Is Winner: #{game_state.winner}"
    puts "Log: #{game_state.log}"
  end
end

# Ejecutar el programa de prueba
TestP1.main
